packages/grid/x-data-grid/src/hooks/features/filter/gridFilterUtils.ts

```diff
@@ -295,36 +296,36 @@ export const buildAggregatedFilterApplier = (
   });
 };
 
+const filterModelItems = defaultMemoize(
```

Comment:
Interestingly, memoization here causes problems after https://github.com/mui/mui-x/pull/9239 was merged.
It's not a pure function, therefore it could return different results for the same inputs.

The [failing test](https://app.circleci.com/pipelines/github/mui/mui-x/39801/workflows/34eb34a5-a202-4fd1-a50e-3aa8a7ed18e4/jobs/228352) indicates the issue - `filterModelItems` does not react to the `columns` change.
Passing `apiRef.current.state.columns` to `filterModelItems` feels hacky, but it fixes the issue and memoization still does its job.
What do you think?

Reply:
I've added a literal cache parameter, it feels less hacky, and as you pointed out in the other discussion memoization isn't really the correct conceptual approach.